src/leveleditor/leveleditor.cpp

```diff
@@ -164,6 +164,13 @@ int				copiedlayer;
 short			x_shake = 0;
 short			y_shake = 0;
 
+int				mouse_x, mouse_y;
+
+void update_mouse_coords() {
+	mouse_x = std::min(640 - 1, std::max(event.button.x, 0));
+	mouse_y = std::min(480 - 1, std::max(event.button.y, 0));
+}
```

Comment:
I switched it from SDL_GetMouseState to event coordinates because it returns values in the 640x480 range (all the code that used event coords didn't break), and it returned values outside this range when the mouse was over the black bars that appear when you have a different aspect ratio.

Reply:
Confirmed the mouse coordinates scaling with the window, nice.